ui/design/src/components/Editor/index.tsx

```diff
@@ -367,87 +368,86 @@ const EditorBox: React.FC<IEditorBox> = React.forwardRef((props, ref) => {
     }
   };
 
-  /**
-   * key handler for the mention popover
-   * inserts the mention on tab, enter or space keypress
-   */
-  const selectMention = (event: KeyboardEvent, mentionRange: Range) => {
-    switch (event.key) {
-      case 'ArrowDown': {
-        event.preventDefault();
-        const prevIndex = index >= slicedMentions.length - 1 ? 0 : index + 1;
-        setIndex(prevIndex);
-        break;
-      }
-      case 'ArrowUp': {
-        event.preventDefault();
-        const nextIndex = index <= 0 ? slicedMentions.length - 1 : index - 1;
-        setIndex(nextIndex);
-        break;
-      }
-      case 'Tab':
-      case 'Enter':
-      case ' ':
-        event.preventDefault();
-        Transforms.select(editor, mentionRange);
-        CustomEditor.insertMention(editor, slicedMentions[index]);
-        setMentionTargetRange(null);
-        break;
-      case 'Escape':
-        event.preventDefault();
-        setMentionTargetRange(null);
-        break;
-    }
-  };
-
-  /**
-   * key handler for the tag popover
-   * inserts the tag on tab, enter keypress
-   * if the user is still on the first position of the popover, on space keypress creates a  new tag
-   * this handles new tag creation when we have tags matching the typed chars
-   */
-  const selectTag = (event: KeyboardEvent, tagRange: Range) => {
-    switch (event.key) {
-      case 'ArrowDown': {
-        event.preventDefault();
-        const prevIndex = index >= slicedTags.length - 1 ? 0 : index + 1;
-        setIndex(prevIndex);
-        break;
-      }
-      case 'ArrowUp': {
-        event.preventDefault();
-        const nextIndex = index <= 0 ? slicedTags.length - 1 : index - 1;
-        setIndex(nextIndex);
-        break;
-      }
-      case 'Tab':
-      case 'Enter':
-        event.preventDefault();
-        Transforms.select(editor, tagRange);
-        CustomEditor.insertTag(editor, slicedTags[index]);
-        setTagTargetRange(null);
-        break;
-      case ' ':
-        if (index === 0 && createTag.length > 1) {
-          event.preventDefault();
-          Transforms.select(editor, tagRange);
-          CustomEditor.insertTag(editor, { name: createTag, totalPosts: 0 });
-        } else {
-          event.preventDefault();
-          Transforms.select(editor, tagRange);
-          CustomEditor.insertTag(editor, slicedTags[index]);
-        }
-        setTagTargetRange(null);
-        break;
-      case 'Escape':
-        event.preventDefault();
-        setTagTargetRange(null);
-        break;
-    }
-  };
-
   const onKeyDown = useCallback(
     event => {
+      /**
+       * key handler for the mention popover
+       * inserts the mention on tab, enter or space keypress
+       */
+      const selectMention = (event: KeyboardEvent, mentionRange: Range) => {
```

Comment:
what is the advantage of having the function declared here? was it because of codefactor complaining?

Reply:
yes, because those methods are used inside the hook only, otherwise they should be listed as dependencies